stripped/colabs/GrupowanieEM_stripped.py

```diff
@@ -0,0 +1,396 @@
+from sklearn.mixture import GaussianMixture
+from sklearn.metrics import accuracy_score, precision_score, jaccard_score, recall_score
+import datashader as ds
+import datashader.transfer_functions as tf
+import colorcet as cc
+import holoviews as hv
+from holoviews.operation.datashader import datashade
+import geoviews as gv
+import geoviews.tile_sources as gts
+from holoviews import opts
+from IPython.display import IFrame
+from IPython.core.display import display
+from bokeh.plotting import show, output_notebook
+
+from pyspark.sql import SparkSession, DataFrame as SparkDataFrame, functions as F
+from pyspark.sql.types import IntegerType, FloatType, StringType, StructType, StructField, Row
+
+from google.colab import drive
+
+spark = (
+         SparkSession.builder
+        .master("local")
+        .appName("Colab")
+        .config('spark.ui.port', '4050')
+        .getOrCreate()
+)
+conf = spark.sparkContext._conf.setAll([('spark.executor.memory', '100g'), ('spark.driver.memory','64g')])
+spark.conf.set("park.driver.maxResultSize", "80g")
+
+spark.conf.set('spark.sql.execution.arrow.enabled', 'true')
+
+drive.mount('/content/drive')
+
+columns = ['lon', 'lat', 'Date', 'Rainf', 'Evap', 'AvgSurfT', 'Albedo','SoilT_40_100cm', 'GVEG', 'PotEvap', 'RootMoist', 'SoilM_100_200cm']
+
+# Utworzenie schematu określającego typ zmiennych
+schema = StructType()
+for i in columns:
+  if i == "Date":
+    schema = schema.add(i, IntegerType(), True)
+  else:
+    schema = schema.add(i, FloatType(), True)
+
+%%time
+# Wczytanie zbioru Nasa w sparku
+
+nasa = spark.read.format('csv').option("header", True).schema(schema).load('/content/drive/MyDrive/BigMess/NASA/NASA.csv')
+
+nasa = (
+    nasa
+    .withColumn('Year', (F.col('Date') / 100).cast('int'))
+    .withColumn('Month', F.col('Date') % 100)
+    .drop('Date')
+)
+nasa.show(5)
+
+nasa.createOrReplaceTempView("nasa")
+
+nasa2020 = spark.sql(''' SELECT * FROM nasa''').where(nasa.Year == 2020).drop('Year')
+nasa2020.show(5)
+
+NASA_sample_annotated = pd.read_csv('/content/drive/MyDrive/BigMess/NASA/NASA_an.csv',sep=';')
+NASA_sample_annotated = spark.createDataFrame(NASA_sample_annotated).withColumnRenamed("lat", "lat_sam").withColumnRenamed("lon", "lon_sam")
+NASA_sample_annotated.show(5)
+
+ NASA_2020_an = nasa2020.join( NASA_sample_annotated,
+                [nasa2020.lon==NASA_sample_annotated.lon_sam , nasa2020.lat==NASA_sample_annotated.lat_sam],
+                "inner").drop('lat_sam').drop('lon_sam')
+
+ NASA_2020_an.show(5)
+
+NASA_2020_set = NASA_2020_an.toPandas()
+
+NASA_2020_set.head(5)
+
+NASA2020_means = (NASA_2020_set[['lon', 'lat', 'Evap','PotEvap','RootMoist','Rainf','SoilM_100_200cm', 'GVEG']].groupby(
+    by=['lon', 'lat']).mean()).reset_index()
+
+#Roczna suma opadów:
+NASA2020_sum = (NASA_2020_set[['lon', 'lat', 'Rainf']].groupby(by=['lon', 'lat']).sum()).rename(columns={"Rainf": "Annual Rainfall"}).reset_index()
+
+#Kolumny ze srednimi miesiecznymi indeksu zielonej roslinnosci:
```

Comment:
Ta sekcja jest z perspektywy kodu zupełnie nieczytelna

Reply:
To znaczy? Bo ja tu tylko agreguję dane i tworzę z nich nowy dataframe... Prosiłabym o jakieś wskazówki, co miałabym poprawić? (bardziej czytelne nazwy zmiennych? ale wtedy nazwy zmiennych będa zapewnie bardzo długie...)

Reply:
Przemyśleć kod i nazewnictwo aby łatwiej było zrozumieć co się dzieje w tym kodzie. Problemem jest czytelność, ale może warto przemyśleć rodzaje zmiennych i sposób składania kodu.

Reply:
Co nieco poprawilam...